docs/topics/lombok.md

```diff
@@ -0,0 +1,137 @@
+[//]: # (title: Lombok compiler plugin)
+
+> The Lombok compiler plugin is [Experimental](components-stability.md). 
+> It may be dropped or changed at any time. Use it only for evaluation purposes. 
+> We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-7112).
+>
+{type="warning"}
+
+The Lombok compiler plugin allows the generation and use of Java's Lombok declarations within the same mixed Java/Kotlin module.
+If you generate these declarations in a Java module and use them in another Kotlin module, 
+then you don't need to use this plugin.
+
+[Lombok](https://projectlombok.org/) generates Java code itself, not the Kotlin Lombok compiler plugin. 
```

Comment:
The beginning is a bit strange: "Lombok generates Java code itself". It's better to be plugin-centric since this is the plugin doc.
Smth like: "The compiler plugin (does what?) while Lombok generates the declarations as it usually does in Java projects."

Reply:
>The compiler plugin (does what?)

He does only what is described above:
>The Kotlin Lombok compiler plugin allows the generation and use of Java's Lombok declarations within the same mixed Java/Kotlin module.